build/build.go

```diff
@@ -249,6 +260,36 @@ func parseAndAugment(pkg *build.Package, isTest bool, fileSet *token.FileSet) ([
 				panic(err)
 			}
 			r.Close()
+
+			if importPath == "runtime" && name == "runtime.go" && vendoredGopherJS != "" {
+				// special case handling for #415: runtime.go contains a string constant named "gopherJSPath" that
+				// contains the import path "github.com/gopherjs/gopherjs/js". If we are using a vendored version of
+				// gopherjs, we must update the contents of the string constant to be the vendored path.
+				for _, decl := range file.Decls {
+					switch d := decl.(type) {
+					case *ast.GenDecl:
+						switch d.Tok {
+						case token.CONST:
+							for _, spec := range d.Specs {
+								for i, name := range spec.(*ast.ValueSpec).Names {
+									if name.Name != "gopherJSPath" {
+										continue
+									}
+									if i < len(spec.(*ast.ValueSpec).Values) {
+										constValue, ok := spec.(*ast.ValueSpec).Values[i].(*ast.BasicLit)
+										if !ok {
+											continue
+										}
+										// update value of constant to be "js" package within vendored gopherjs
+										constValue.Value = fmt.Sprintf(`"%s/js"`, vendoredGopherJS)
+									}
+								}
+							}
+						}
+					}
+				}
+			}
```

Comment:
Yes, agree that it would be better to decomp this function (just wanted to verify that the over-all approach was valid before investing extra time in refactoring)